backend/schemas/credit.py

```
@@ -1,31 +1,71 @@
+        if isinstance(v, str):
+            # Validate that it's a proper ISO-8601 format
+            try:
+                datetime.fromisoformat(v.replace('Z', '+00:00'))
+                return v
+            except ValueError:
+                # If it's not a valid ISO string, try to parse and convert
+                try:
+                    parsed = datetime.fromisoformat(v)
+                    return parsed.isoformat()
+                except ValueError:
+                    raise ValueError(f"Invalid datetime format: {v}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)